### PR TITLE
Care gaps components

### PR DESCRIPTION
**Pull request by sakhin123**
## Summary
Implementation of Care Gaps button on Measure home page and Care Gaps page render with input components. 

**NOTE:** There are placeholders (anything under the Program text input component) rendered for formatting purposes that are not part of the task and need not be evaluated at this time. Some state variables are also used but not set as a result.

## New behavior
- A "Calculate Care Gaps" button is rendered on any Measure's home page (in between the "Evaluate Measure" and "Delete" buttons).
- If the "Calculate Care Gaps" button is pressed, user is redirected to the /Measure/[id]/careGaps page. 
- Where the following are rendered: 2 pre-filled DatePickers, 2 radio buttons (Select, Organization), AutoComplete components for Select Patient, Organization, and Practitioner, TextInput component for Program, and Placeholders for the request preview and calculate button

## Code changes
- **pages/[resource]/[id]/careGaps.tsx** added, renders the new page and components
- **__tests__/pages/resource/id/careGaps.tsx** added with unit tests

## Testing Guidance
- See the README.md for first time setup instructions.
#### Test in a browser
- To start the frontend: `npm run dev` then navigate to http://localhost:3001 in a browser
1. Valid navigation:
    - Click on “Measure” from the left-hand-side navigation menu. This will load the measure resource ID page. Click on any of the IDs. 
    - Click the “Calculate Care Gaps” button that is at the top of the main page body and to the left of the “Delete” button. User should be redirected to /Measure/[id]/careGaps page, where the following is rendered: 2 pre-filled DatePickers, 2 radio buttons (Select, Organization), AutoComplete components for Select Patient, Organization, and Practitioner, TextInput component for Program, and Placeholders for the request preview and calculate button.
      - The Subject radio button should be selected by default, and the Select Patient AutoComplete component will be required. The Select Organization and Select Practitioner components will be disabled.
      - Select the Organization radio button. Then the Select Organization and Select Practitioner components will be enabled (Select Organization will be required). The Select Patient component will be disabled.
    - User will be able to type into any enabled text/DatePicker components
2. Invalid navigation (non-Measure resource): 
    - Navigate to any [resourceType]/[id] page for a non-Measure resource. Then add "careGaps" to the end of the url and hit enter. An error message should be displayed on the main page body:  "Cannot calculate care gaps on resourceType: [resourceType], only on resourceType: Measure"

#### Unit tests
- Run the unit tests: `npm run test`

### CODE REVIEW

File: __tests__/pages/resource/id/careGaps.test.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        id: "numer-EXM125-3",
+        meta: {
+          profile: [
+            "http://hl7.org/fhir/us/core/StructureDefinition/us-core-practitionerreport-note",
+          ],
+        },
+      },
+    },
+  ],
+};
```
Comment (elsaperelli) [inline __tests__/pages/resource/id/careGaps.test.tsx:58]: I have seen that this is in a couple of test files and it is pretty long so @mgramigna would it be okay/beneficial for them to pull this out into it's own file?

**Review by elsaperelli — CHANGES_REQUESTED** [review]
Great work! Just a few comments- some nit-picky, some just questions :)

File: pages/[resourceType]/[id]/careGaps.tsx [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+const DEFAULT_PERIOD_END = new Date(`${DateTime.now().year}-12-31T00:00:00`);
+
+const CareGapsPage = () => {
+  const router = useRouter();
+  const { resourceType, id } = router.query;
+  const [radioValue, setRadioValue] = useState("Subject");
+  const [fetchingError, setFetchingError] = useState(false);
+  const [loadingRequest, setLoadingRequest] = useState(false);
+  const [measureReportBody, setMeasureReportBody] = useState("");
+  const [gridColSpans, setGridColSpans] = useState([3, 3, 0]);
```
[the remark this answers is not in the document]
Comment (sakhin123) [reply pages/[resourceType]/[id]/careGaps.tsx:45]: Yes, all these state variables will be fully used in the next PR, to follow the same rendering pattern as in evaluate measure. I basically re-used code from evaluate.tsx which fully implements these state variables and conditional rendering. I can remove them but I would have to re-add them for the next task.

File: pages/[resourceType]/[id]/index.tsx [hunk 3]
```diff
@@ -70,22 +70,40 @@ function ResourceIDPage() {
...
-              float: "right",
-              marginRight: "8px",
-              marginLeft: "8px",
-            }}
-          >
-            <div>Evaluate Measure</div>
-          </Button>
-        </Link>
+        <div>
+          <Link href={`/${resourceType}/${id}/careGaps`} key={`care-gaps-${id}`} passHref>
```
Comment (elsaperelli) [inline pages/[resourceType]/[id]/index.tsx:74]:
```suggestion
          <Link href={`/${resourceType}/${id}/care-gaps`} key={`care-gaps-${id}`} passHref>
```
Matt informed me that routes usually don't have camelCase.

**Review by elsaperelli — CHANGES_REQUESTED** [review]
One more change that I forgot about but other than that this is almost ready to go!

File: pages/[resourceType]/[id]/care-gaps.tsx [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+  const [periodStart, setPeriodStart] = useState<Date>(DEFAULT_PERIOD_START);
+  const [periodEnd, setPeriodEnd] = useState<Date>(DEFAULT_PERIOD_END);
+  useEffect(() => {
+    if (radioValue === "Subject") {
+      setOrganizationValue("");
+      setPractitionerValue("");
+    } else if (radioValue === "Organization") {
+      setPatientValue("");
+    }
+  }, [radioValue]);
```
Comment (mgramigna) [inline pages/[resourceType]/[id]/care-gaps.tsx:58]: love this

File: __tests__/pages/resource/id/careGaps.test.tsx (hunk 1 above)
Comment (mgramigna) [reply __tests__/pages/resource/id/careGaps.test.tsx:58]: no preference, fine either way

File: pages/[resourceType]/[id]/care-gaps.tsx [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+        <div>
+          <BackButton />
+          <Center>
+            <h2 style={{ color: textGray, marginTop: "0px", marginBottom: "4px" }}>
+              Gaps in Care: {id}
+            </h2>
+          </Center>
+          <Divider my="md" />
+          <Grid columns={gridColSpans[0]}>
+            <MantineProvider
```
[the remark this answers is not in the document]
Comment (sakhin123) [reply pages/[resourceType]/[id]/care-gaps.tsx:73]: sometimes we have a component specific color scheme we try to override, that's why there are a few MantineProviders throughout the application

**Review by elsaperelli — APPROVED** [review]
This is great work!

**Pull request merged**